src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1217,94 +1140,107 @@ protected void shutdownService() {
       this.runningThread = null;
       runningThread.interrupt();
     }
-    
-    /**
-     * Called when the queue has been updated and we may need to wake up the consumer thread.
-     */
-    protected void handleQueueUpdate() {
-      if (! startIfNotStarted()) {
-        Thread currRunningThread = runningThread;
-        if (currRunningThread != null) {
-          LockSupport.unpark(currRunningThread);
-        }
-      }
-    }
 
     @Override
     public void run() {
+      Worker worker = null;
       while (runningThread != null) {
         try {
-          TaskWrapper nextTask = getNextTask();
-          if (nextTask != null) {
-            switch (queuePriority) {
-              case High:
-                workerPool.runHighPriorityTask(nextTask);
-                break;
-              case Low:
-                workerPool.runLowPriorityTask(nextTask);
-                break;
-              default:
-                throw new UnsupportedOperationException();
+          worker = workerPool.getWorker();
```

Comment:
Here is where we get the worker, then the next task, then provide the task to the worker...and loop.